File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_ExportCau_Causali/Models/par-export-cau-causali-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";

export class Par_ExportCau_CausaliModel {
  id: number;
  idPar_ExportCau: number;
  idCausale: number;
  codice: string;
  tipoElaborazione: Par_Export_TipoElaborazione;
  tipoUnita: Par_Export_TipoUnita;
}

export enum Par_Export_TipoElaborazione {
  Gionaliera = 0,
  Mensile    = 1
}

export enum Par_Export_TipoUnita {
  Ore     = 0,
  Giorni  = 1,
  Importo = 2
}

export class Par_ExportCau_Causali_Get_InModel {
  id: number = 0; // id del Par_ExportCau padre
}
export class Par_ExportCau_Causali_Get_OutModel extends ModelResult {
  par_ExportCau_Causali: Par_ExportCau_CausaliModel[] = [];
}

export class Par_ExportCau_Causali_Put_InModel {
  idPar_ExportCau: number = 0;
  par_ExportCau_Causali: Par_ExportCau_CausaliModel[] = [];
}
export class Par_ExportCau_Causali_Put_OutModel extends ModelResult {
  par_ExportCau_Causali: Par_ExportCau_CausaliModel[] = [];
}

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_ExportCau_CausaliService/Models/Par_ExportCau_CausaliModel.cs
using nvxapp.server.data.Entities.Tenant.GestionePresenze;
using nvxapp.server.service.ClientServer_Service.ModelsBase;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_ExportCau_CausaliService.Models
{
    public class Par_ExportCau_CausaliModel
    {
        public int Id { get; set; }
        public int IdPar_ExportCau { get; set; }
        public int IdCausale { get; set; }
        public string? Codice { get; set; }
        public Par_Export_TipoElaborazione TipoElaborazione { get; set; }
        public Par_Export_TipoUnita TipoUnita { get; set; }     
    }

    public class Par_ExportCau_Causali_GetAll_InModel
    {
        public int IdPar_ExportCau { get; set; }
    }
    public class Par_ExportCau_Causali_GetAll_OutModel : ModelResult
    {
        public List<Par_ExportCau_CausaliModel> Par_ExportCau_Causali { get; set; } = new();
    }

    public class Par_ExportCau_Causali_Get_InModel
    {
        public int Id { get; set; }
    }
    public class Par_ExportCau_Causali_Get_OutModel : ModelResult
    {
        public List<Par_ExportCau_CausaliModel> Par_ExportCau_Causali { get; set; } = new();
    }

    public class Par_ExportCau_Causali_Put_InModel
    {
        public int IdPar_ExportCau { get; set; }
        public List<Par_ExportCau_CausaliModel> Par_ExportCau_Causali { get; set; } = new();
    }
    public class Par_ExportCau_Causali_Put_OutModel : ModelResult
    {
        public List<Par_ExportCau_CausaliModel> Par_ExportCau_Causali { get; set; } = new();
    }

    public class Par_ExportCau_Causali_Delete_InModel
    {
        public int Id { get; set; }
    }
    public class Par_ExportCau_Causali_Delete_OutModel : ModelResult { }
}
